ingestion_tools/scripts/data_validation/tests/test_tilt_angles.py

```diff
@@ -0,0 +1,256 @@
+import os
+from typing import Dict, List
+
+import numpy as np
+import pandas as pd
+import pytest
+
+ANGLE_TOLERANCE = 0.05
+
+
+@pytest.mark.tilt_angles
+@pytest.mark.metadata
+@pytest.mark.parametrize("run_name", pytest.run_name, scope="session")
+class TestTiltAngles:
+    """
+    A class to test tilt angle data. Only checking that the # of angles in these files are consistent with other files /
+    data and that they properly map to other files / data. Other data validation tests is done in respective classes.
+    Spans .tlt, .rawtlt, .mdoc, tiltseries_metadata.json, frames files.
+    Ordering:
+        - .tlt (<=) maps to .rawtlt (not necessarily 1:1)
+        - .rawtlt (<=) maps to .mdoc (not necessarily 1:1)
+        - .mdoc (==) one-to-one with frames files & tiltseries_metadata size["z"]
+        - tiltseries metadata size["z"] == frames_count == # of frames files
+        - # of frames_files <= metadata tilt_range (see max_frames_count fixture)
+
+    Extra test redundancy is added for the cases where files sometimes do not exist.
+    Extra redundancy tests are done on the tiltseries metadata size["z"] since frames_count / # of frames files may be 0
+        when no frames are present and that is a valid case. To ensure consistency, we rely on the check
+        tiltseries metadata size["z"] == frames_count == # of frames files.
+    """
+
+    @pytest.fixture(autouse=True)
+    def max_frames_count(self, tiltseries_metadata: Dict):
+        # TODO FIXME how to check if -0.0 should exist
+        return (
+            tiltseries_metadata["tilt_range"]["max"] - tiltseries_metadata["tilt_range"]["min"]
+        ) / tiltseries_metadata["tilt_step"] + 1
+
```

Comment:
This might not always hold true, as there is a possibility of frames that were generated not being used for the tiltseries.

Reply:
Sounds good, I'll remove all the max_frames_count checks in this PR and the future ones.